### DISCUSSION TITLE
Join Tables on client side and put in Perspective Viewer?

### DISCUSSION THREAD

**Discussion by nekw** [perspective-dev/perspective#1509] [Q&A] [answered]
In client side e.g. JavaScript client, is there a way to get data from 2 perspective server tables and then join them on client side and put back in Perspective Viewer to display? Thanks.

**Comment by texodus** [comment, marked as answer]
You can effect simple outer joins with Perspective's [partial updates](https://perspective.finos.org/docs/md/table.html#index-and-limit) feature,  check out the [NYC Citibike](https://bl.ocks.org/texodus/bc8d7e6f72e09c9dbd7424b4332cacad#citibike.js) example for this.

Ad-hoc joins of already instantiated `Table`s are a a work-in-progress.
